src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import com.tinkerpop.blueprints.Graph
+import com.tinkerpop.blueprints.impls.orient.OrientConfigurableGraph.THREAD_MODE
+import com.tinkerpop.blueprints.impls.orient.{OrientGraph, OrientGraphFactory}
+
+object DataSource {
+  def apply(url: String, user: String, password: String, minPoolSize: Int, maxPoolSize: Int) = {
+    val factory: OrientGraphFactory = createFactory(url, user, password)
+    factory.setupPool(minPoolSize, maxPoolSize)
+    new DataSource(factory)
+  }
+
+  def apply(url: String, user: String, password: String) = {
+    val factory: OrientGraphFactory = createFactory(url, user, password)
+    new DataSource(factory)
+  }
+
+  def createFactory(url: String, user: String, password: String): OrientGraphFactory = {
+    val factory = new OrientGraphFactory(url, user, password)
+    factory.setThreadMode(THREAD_MODE.MANUAL)
+    factory.setAutoStartTx(false)
+    factory
+  }
+}
+
+class DataSource(graphFactory: OrientGraphFactory) {
+  def openConnection() = new DataSourceConnection(graphFactory.getTx)
+  def shutdown() = graphFactory.close()
+}
+
+class DataSourceConnection(graph: OrientGraph) {
```

Comment:
It only does happen that way in the code, but should the DAOs get a different class with just the ```withGraph``` method and any other functions the DAO needs but not this full functionality?